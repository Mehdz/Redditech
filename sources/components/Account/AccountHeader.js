import React from 'react';
import {ImageBackground, Platform, StyleSheet, View} from 'react-native';
import {Avatar, Caption, Title} from 'react-native-paper';
import {PropTypes} from 'prop-types';

const AccountHeader = ({data}) => {
    return (
        <ImageBackground
            style={styles.containerBanner}
            blurRadius={15}
            source={{
                uri:
                    data &&
                    data.subreddit &&
                    data.subreddit.banner_img &&
                    data.subreddit.banner_img.split('?')[0],
            }}
            resizeMode="cover">
            <View style={styles.BannerFilter}>
                <Avatar.Image
                    size={100}
                    style={styles.containerAvatar}
                    source={{
                        uri:
                            data &&
                            data.icon_img &&
                            data.icon_img.split('?')[0],
                    }}
                />
                <View>
                    <Title style={styles.titlePosts}>
                        {data && data.comment_karma}
                    </Title>
                    <Caption style={styles.captionPosts}>
                        Comment{'\n'}Karma
                    </Caption>
                </View>
                <View>
                    <Title style={styles.titlePosts}>
                        {data && data.link_karma}
                    </Title>
                    <Caption style={styles.captionPosts}>
                        Posts{'\n'}Karma
                    </Caption>
                </View>
            </View>
        </ImageBackground>
    );
};

const styles = StyleSheet.create({
    containerBanner: {
        flex: Platform.OS === 'ios' ? 0.5 : 0.4,
    },
    BannerFilter: {
        flex: 1,
        backgroundColor: 'rgba(255, 0, 0, 0.3)',
        alignItems: 'center',
        padding: 40,
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    containerAvatar: {
        marginTop: Platform.OS === 'ios' ? 30 : 0,
    },
    titlePosts: {
        fontSize: 30,
        textAlign: 'center',
        color: 'white',
        textShadowColor: '#585858',
        textShadowOffset: {width: 0, height: 1},
        textShadowRadius: 15,
        marginTop: Platform.OS === 'ios' ? 50 : 0,
    },
    captionPosts: {
        textAlign: 'center',
        fontSize: 20,
        color: 'white',
        textShadowColor: '#585858',
        textShadowOffset: {width: 0, height: 1},
        textShadowRadius: 15,
    },
});

AccountHeader.propTypes = {
    data: PropTypes.object.isRequired,
};

export default AccountHeader;
