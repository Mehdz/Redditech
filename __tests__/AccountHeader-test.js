// __tests__/AccountHeader-test.js
import React from 'react';
import renderer from 'react-test-renderer';
import AccountHeader from '../sources/components/Account/AccountHeader';

test('renders correctly', () => {
  const tree = renderer.create(<AccountHeader />).toJSON();
  expect(tree).toMatchSnapshot();
});