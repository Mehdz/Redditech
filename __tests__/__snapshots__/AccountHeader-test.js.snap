// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 0.5,
    }
  }
>
  <Image
    blurRadius={15}
    resizeMode="cover"
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 0, 0, 0.3)",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 40,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#6200ee",
            "borderRadius": 50,
            "height": 100,
            "width": 100,
          },
          Object {
            "marginTop": 30,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "borderRadius": 50,
            "height": 100,
            "width": 100,
          }
        }
      />
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "500",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 20,
                  "letterSpacing": 0.15,
                  "lineHeight": 30,
                  "marginVertical": 2,
                },
                Object {
                  "color": "white",
                  "fontSize": 30,
                  "marginTop": 50,
                  "textAlign": "center",
                  "textShadowColor": "#585858",
                  "textShadowOffset": Object {
                    "height": 1,
                    "width": 0,
                  },
                  "textShadowRadius": 15,
                },
              ],
            ],
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontFamily": "System",
                "fontWeight": "400",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 12,
                  "letterSpacing": 0.4,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "textAlign": "center",
                  "textShadowColor": "#585858",
                  "textShadowOffset": Object {
                    "height": 1,
                    "width": 0,
                  },
                  "textShadowRadius": 15,
                },
              ],
            ],
          ]
        }
      >
        Comment
        

        Karma
      </Text>
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "500",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 20,
                  "letterSpacing": 0.15,
                  "lineHeight": 30,
                  "marginVertical": 2,
                },
                Object {
                  "color": "white",
                  "fontSize": 30,
                  "marginTop": 50,
                  "textAlign": "center",
                  "textShadowColor": "#585858",
                  "textShadowOffset": Object {
                    "height": 1,
                    "width": 0,
                  },
                  "textShadowRadius": 15,
                },
              ],
            ],
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontFamily": "System",
                "fontWeight": "400",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 12,
                  "letterSpacing": 0.4,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "textAlign": "center",
                  "textShadowColor": "#585858",
                  "textShadowOffset": Object {
                    "height": 1,
                    "width": 0,
                  },
                  "textShadowRadius": 15,
                },
              ],
            ],
          ]
        }
      >
        Posts
        

        Karma
      </Text>
    </View>
  </View>
</View>
`;
